frontend/components/SetupPage/FundYourAgent/hooks/useGetRefillRequirementsWithMonthlyGas.ts

```diff
@@ -0,0 +1,198 @@
+  getNativeTokenSymbol,
+  TOKEN_CONFIG,
+  TokenConfig,
+} from '@/config/tokens';
+import { AddressZero } from '@/constants/address';
+import { EvmChainId } from '@/constants/chains';
+import { SERVICE_TEMPLATES } from '@/constants/serviceTemplates';
+import { TokenSymbolConfigMap, TokenSymbolMap } from '@/constants/token';
+import { useBalanceAndRefillRequirementsContext } from '@/hooks/useBalanceAndRefillRequirementsContext';
+import { useMasterWalletContext } from '@/hooks/useWallet';
+import { Address } from '@/types/Address';
+import { AgentConfig } from '@/types/Agent';
+import { bigintMax } from '@/utils/calculations';
+import { formatUnitsToNumber } from '@/utils/numberFormatters';
+
+import { useBeforeBridgeFunds } from '../../Create/SetupEoaFunding/useBeforeBridgeFunds';
+import { TokenRequirement } from '../TokensRequirements';
+
+const ICON_OVERRIDES: Record<string, string> = {
+  [TokenSymbolMap['XDAI']]: '/tokens/wxdai-icon.png',
+};
+
+const getTokenMeta = (
+  tokenAddress: Address,
+  chainConfig: ChainTokenConfig,
+  evmHomeChainId: EvmChainId,
+) => {
+  if (tokenAddress === AddressZero) {
+    const symbol = getNativeTokenSymbol(evmHomeChainId);
+    const config = chainConfig[symbol];
+    return {
+      symbol,
+      decimals: config.decimals,
+      iconSrc: TokenSymbolConfigMap[symbol].image || '',
+    };
+  }
+
+  const tokenDetails = getTokenDetails(
+    tokenAddress,
+    chainConfig,
+  ) as TokenConfig;
+
+  return {
+    symbol: tokenDetails.symbol,
+    decimals: tokenDetails.decimals,
+    iconSrc: TokenSymbolConfigMap[tokenDetails.symbol].image || '',
+  };
+};
+
+const getTokensDetailsForFunding = (
+  requirementsPerToken: { [tokenAddress: Address]: string },
+  evmHomeChainId: EvmChainId,
+  chainConfig: ChainTokenConfig,
+) => {
+  const tokenRequirements: TokenRequirement[] = Object.entries(
+    requirementsPerToken,
+  )
+    .map(([tokenAddress, amount]) => {
+      const { symbol, decimals, iconSrc } = getTokenMeta(
+        tokenAddress as Address,
+        chainConfig,
+        evmHomeChainId,
+      );
+      const parsedAmount = formatUnitsToNumber(amount, decimals);
+
+      if (parsedAmount > 0) {
+        return {
+          amount: parsedAmount,
+          symbol,
+          iconSrc: ICON_OVERRIDES[symbol] || iconSrc,
+        };
+      }
+    })
+    .filter(Boolean) as TokenRequirement[];
+
+  return tokenRequirements.sort((a, b) => b.amount - a.amount);
+};
+
+/**
+ * Hook to get the refill requirements for the selectedAgent — considers the monthly_gas_estimate
+ * in order to evaluate the requirements.
+ * example output:
+ * {
+ *   tokenRequirements: [
+ *     {
+ *       amount: 0.5,
+ *       symbol: "XDAI",
+ *       iconSrc: "/tokens/wxdai-icon.png"
+ *     },
+ *     {
+ *       amount: 100,
+ *       symbol: "USDC",
+ *       iconSrc: "/tokens/usdc-icon.png"
+ *     }
+ *   ],
+ *   isLoading: false
+ * }
+ */
+export const useGetRefillRequimentsWithMonthlyGas = ({
```

Comment:
1) can this be anyhow used in/instead of this hook `useGetBridgeRequirementsParamsWithMonthlyGasEstimate`?
2) should this hook also have the same warnings that it should not exist? :) cc @mohandast52

Reply:
1. We may merge them later, `useGetBridgeRequirementsParamsWithMonthlyGasEstimate` supports only native token as of now
2. Will add the warning